forgevtt-module.js

```diff
@@ -2322,11 +2322,13 @@ class ForgeVTT_FilePicker extends FilePicker {
         if (!ForgeVTT.usingTheForge && source !== "forgevtt")
             return super.upload(source, target, file, body, { notify }); //in v8, body will be the options.
 
-        // Some uploads e.g. dragging an image onto journal have no target but have a UUID. Upload to the active world folder
+        // Some uploads e.g. dragging an image onto journal have no target but have a UUID.
+        // body.uuid is the UUID of the parent document that the entity is being uploaded to, not the file itself.
+        // Upload to the active world folder with uuid and timestamp as the target if no target is provided.
         if (target == null) {
             target = "";
             if (body.uuid) {
-                target = `worlds/${game.world.id || game.world.name}`
+                target = `worlds/${game.world.id || game.world.name}/assets/${body.uuid}/${Date.now()}`;
```

Comment:
I'd recommend using a `YYYY-MM-DD HH:SS` folder name instead of epoch time. Will make it more user friendly

Reply:
Some of those, like colons, might not be safe on all file systems and could cause issues on an asset sync
I've opted for a readable variation on `new Date().toISOString()` that gives us the unique millisecond (just using seconds might not be enough if people drag quickly or upload programmatically)

```js
// Get the ISO string and replace characters that are not suitable for folder names
var uniqueId = new Date().toISOString();
// Replace characters that might cause issues in file systems, and remove Z at the end
uniqueId = uniqueId.replace(/:/g, "-").replace("T", "_").slice(0, -1);
target = `worlds/${game.world.id || game.world.name}/assets/${body.uuid}/${uniqueId}`;
```

The new code would give us `2024-04-03_21-27-24.579`